### PR TITLE
Added jit log stream changing function and some refactor

### LINKED ISSUE #54182: [JIT] Provide an API to choose a different output stream for `PYTORCH_JIT_LOG_LEVEL`
## 🚀 Feature
Provide an API to choose a different output stream for `PYTORCH_JIT_LOG_LEVEL`, which by default is stderr.

## Motivation

The current default (stderr) makes it cumbersome to turn on logging in python notebooks. An ability to use stdout or another output stream should help with this issue.

## Code Pointers

The relevant code is located in two files:
 * [torch/csrc/jit/jit_log.h](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/jit_log.h)
 * [torch/csrc/jit/jit_log.cpp](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/jit_log.cpp)

cc @gmagogsfm, @Krovatkin

### PR DESCRIPTION
**Pull request by narang99**
Description:
- Have only added `stdout` and `stderr` as possible options from python
  API for now. We can do file path passing later maybe.
- Put the class `JitLoggingConfig` in the cpp file as none of its methods were being used outside of this file.

Python API:
`torch._C._jit_set_logging_stream('stdout|stderr')`
C++ API:
`::torch::jit::set_jit_logging_output_stream(ostream);`

Testing:
- Tested python API locally.
- Unit test for the C++ API is written

Fixes #54182

### CODE REVIEW

**Comment by narang99** [comment]
I'm stumped by the windows build error, it would be good if you guys could take a look and help me if thats possible. (I dont have a windows machine so I have to keep using the CI for the build)

File: torch/csrc/jit/python/init.cpp [hunk 1]
```diff
@@ -667,6 +667,21 @@ void initJITBindings(PyObject* module) {
...
+      .def(
+          "_jit_set_logging_stream",
+          [](std::string stream_name) -> void {
+            if(stream_name == "stdout") {
+              ::torch::jit::set_jit_logging_output_stream(std::cout);
+            }
+            else if(stream_name == "stderr")  {
+              ::torch::jit::set_jit_logging_output_stream(std::cerr);
+            }
+            else {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/python/init.cpp:679]: Do you mind running clang-format on the PR? You can use `tools/linter/clang_format_all.py` - it downloads a proper version of clang-format and runs it on the entire codebase (if it reformats files not touched by my changes, I usually undo those reformats)

File: torch/csrc/jit/jit_log.h [hunk 2]
```diff
@@ -49,45 +50,14 @@ enum class JitLoggingLevels {
...
-  }
-};
-
 std::string TORCH_API get_jit_logging_levels();
 
 void TORCH_API set_jit_logging_levels(std::string level);
 
+void TORCH_API set_jit_logging_output_stream(std::ostream& out_stream);
+
+TORCH_API std::ostream& get_jit_logging_output_stream();
```
Comment (ZolotukhinM) [inline torch/csrc/jit/jit_log.h:59]: Nit: it probably doesn't raise a compiler error, but for consistency let's put TORCH_API after the return type.

File: torch/csrc/jit/jit_log.cpp [hunk 3]
```diff
@@ -26,6 +71,14 @@ void TORCH_API set_jit_logging_levels(std::string level) {
   JitLoggingConfig::getInstance().setLoggingLevels(level);
 }
 
+void TORCH_API set_jit_logging_output_stream(std::ostream& stream) {
+  JitLoggingConfig::getInstance().setOutputStream(stream);
+}
+
+TORCH_API std::ostream& get_jit_logging_output_stream()  {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/jit_log.cpp:78]: I think TORCH_API is only needed at declaration of a function, the definition doesn't need it.

**Comment by ZolotukhinM** [comment]
Thanks for the change, it mostly looks good! I wonder if these failures are real, do you mind trying to rebase your change on `origin/viable/strict`? This branch slightly lags behind `origin/master`, but usually is more stable (it's advanced only to commits that pass certain testing).

File: torch/csrc/jit/jit_log.cpp (hunk 3 above)
Comment (narang99) [reply torch/csrc/jit/jit_log.cpp:78]: I pushed the change again because it was giving windows build errors. It couldnt find the symbol.

**Comment by narang99** [comment]
Closing this in favor of [this PR](https://github.com/pytorch/pytorch/pull/65768)

**Pull request closed without merging**